ixfr.c

```diff
@@ -1443,6 +1432,22 @@ int ixfr_store_delrr_uncompressed(struct ixfr_store* ixfr_store,
 		&ixfr_store->data->del_len, &ixfr_store->del_capacity);
 }
 
+static size_t skip_dname(uint8_t* rdata, size_t rdata_len)
+{
+	for (size_t index=0; index < rdata_len; ) {
+		uint8_t label_size = rdata[index];
+		if (label_size == 0) {
+			return index + 1;
+		} else if ((label_size & 0xc0) != 0) {
+			index += 1;
+		} else {
+			index += label_size;
+		}
+	}
```

Comment:
Perhaps this change is not optimal as the for condition already checks for it. It could be cautious, with more prominence on the boundary check, in the code, than it has currently. If you like it, perhaps.

Reply:
I'd like to keep this one as-is, but I did add a comment to clarify that if `index` exceeds `rdata_len`, `0` is automatically returned as the condition breaks the loop.

Reply:
Perhaps there is another issue here, when there is a compression pointer it attempts to continue the loop. This is not really good, because it does not want to follow the dname compression pointer, it is attempting to skip the dname that is at this point, and so it should return after moving ahead one byte. So check if there is another byte, then skip that byte, and then return from the function, much like in the zero label case, since the dname has ended, in a compression pointer?

Reply:
Clearly I don't understand dname compression yet :sweat_smile: Well, we already know that there's another byte available as `index < rdata_len`, so that'd mean simply `return index + 1;` like with the zero label?

Reply:
or:
```
static size_t skip_dname(uint8_t* rdata, size_t rdata_len)
{   
  for (size_t index=0; index < rdata_len; ) {
    uint8_t label_size = rdata[index];
    if (label_size == 0 || (label_size & 0xc0) != 0) {
      return index + 1;
    } else {
      /* loop breaks if index exceeds rdata_len */
      index += label_size;
    }
  }
  
  return 0;
}
```

Reply:
No, because it is one byte longer, check if that byte is in the rdata_len? The dname compression pointer is two bytes.

Reply:
so:
```c
static size_t skip_dname(uint8_t* rdata, size_t rdata_len)
{
  for (size_t index=0; index < rdata_len; ) {
    uint8_t label_size = rdata[index];
    if (label_size == 0) {
      return index + 1;
    } else if ((label_size & 0xc0) != 0) {
      return (index + 1 < rdata_len) ? index + 2 : 0;
    } else {
      /* loop breaks if index exceeds rdata_len */
      index += label_size;
    }
  }

  return 0;
}
```

Reply:
As the changes were already merged, I pushed the fix to master directly.